### PR TITLE
chore (build): Update to use pnpm v9.12.3.

### PR DESCRIPTION
**Pull request by shaper**
v9 has been out since approximately April 2024 and is compatible with Node v18 and v20 per:

https://pnpm.io/installation#compatibility

Note the SDK CI is today running v8.6.9, thus this change also updates the GitHub CI action to the newer version.

### CODE REVIEW

**Comment by lgrammel** [comment]
The Sveltekit error looks legit. It could be that the pnpm 8.6 builds swallowed the error for some reason. The best path forward would imo be to attempt to fix the Sveltekit error.

File: examples/next-openai/app/assistant/page.tsx [hunk 1]
```diff
@@ -8,6 +8,8 @@ const roleToColorMap: Record<Message['role'], string> = {
   user: 'black',
   assistant: 'green',
   data: 'orange',
+  function: 'pink',
+  tool: 'blue',
```
Comment (lgrammel) [inline examples/next-openai/app/assistant/page.tsx:12]: revert

File: examples/next-openai/app/use-chat-throttle/page.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ export default function Chat() {
   const { messages, input, isLoading, error, handleInputChange, handleSubmit } =
     useChat({
       api: '/api/use-chat-throttle',
-      experimental_throttle: 50,
+      // experimental_throttle: 50,
```
Comment (lgrammel) [inline examples/next-openai/app/use-chat-throttle/page.tsx:15]: revert

File: examples/next-openai/app/use-completion-throttle/page.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ export default function Chat() {
 
   const { completion, input, handleInputChange, handleSubmit } = useCompletion({
     api: '/api/use-completion-throttle',
-    experimental_throttle: 50,
+    // experimental_throttle: 50,
```
Comment (lgrammel) [inline examples/next-openai/app/use-completion-throttle/page.tsx:14]: revert

**Comment by shaper** [comment]
Status: revisit after typescript upgrade.

**Comment by lgrammel** [comment]
it's the `latest` imports that are breaking. need to switch to fixed versions again

**Review by lgrammel — APPROVED** [review]

**Pull request merged**